axum-macros/src/typed_path.rs

```diff
@@ -314,6 +372,33 @@ enum Segment {
     Static(String),
 }
 
+fn path_rejection() -> TokenStream {
+    quote! {
+        <::axum::extract::Path<Self> as ::axum::extract::FromRequest<B>>::Rejection
+    }
+}
+
+fn rejection_assoc_type(rejection: &Option<syn::Path>) -> TokenStream {
+    if let Some(rejection) = rejection {
+        quote! { #rejection }
+    } else {
+        path_rejection()
+    }
+}
+
+fn map_err_rejection(rejection: &Option<syn::Path>) -> TokenStream {
+    if let Some(rejection) = rejection {
+        let path_rejection = path_rejection();
+        quote! {
+            .map_err(|rejection| {
+                <#rejection as ::std::convert::From<#path_rejection>>::from(rejection)
+            })
+        }
+    } else {
+        quote! {}
+    }
```

Comment:
`Option<TokenStream>` can also be interpolated, so this can be

```suggestion
    rejection.map(|rejection| {
        let path_rejection = path_rejection();
        quote! {
            .map_err(|rejection| {
                <#rejection as ::std::convert::From<#path_rejection>>::from(rejection)
            })
        }
    })
```

(or you can keep the function signature and add `.unwrap_or_default()` to that snippet)